src/Http/ServerRequest.php

```diff
@@ -517,6 +539,34 @@ public function is(array|string $type, mixed ...$args): bool
         return $this->_detectorCache[$type] = $this->_detectorCache[$type] ?? $this->_is($type, $args);
     }
 
+    /**
+     * Returns a piece of data from the Request converted to a given data type.
+     *
+     * Example: `asType('Int', $this->params, 'id', 0)` will return
+     * the 'id' parameter from the request converted to an integer.
+     *
+     * @param string $type The desired type to convert the data to. Example: 'Int', 'Bool', 'String'.
```

Comment:
Is there a way in phpstan/psalm to have a overload for literal values and their narrowed types? For example 'Int' has a return type of `int`.

Reply:
I was researching `@psalm-return` annotations but for the amount of types this use case needs the annotation quickly grew very large (10+ lines). This is what I started off with:
```
/**
   * @template T of int|string|bool|float|\Cake\I18n\Date|\Cake\I18n\DateTime
   * @param T $id
   * @psalm-return (
   *     T is int
   *     ? static
   *     : array<static>
   * ) 
   */
```
I was testing in the [psalm sandbox](https://psalm.dev/)